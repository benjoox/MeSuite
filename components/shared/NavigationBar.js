import React, { useContext } from 'react'
import { useAuth0 } from "@auth0/auth0-react"
import Link from 'next/link'
import { Navbar, Nav, Spinner } from 'react-bootstrap'
import UserMenu from './UserMenu'
import DataMode from '../portfolio/Transaction/DataMode'

export default function NavigationBar() {
    const { isLoading } = useAuth0()

    return (
            <Navbar bg="dark" expand="lg" className="d-flex">
                <Link href='/' passHref> 
                    <Nav.Link> 
                        <h3>Me Suite</h3>
                    </Nav.Link> 
                </Link>
                <Navbar.Toggle aria-controls="navbar-nav" />
                <Navbar.Collapse id="navbar-nav" className="d-flex justify-content-end">
                    <div>
                        <DataMode />
                    </div>
                    <Link href='/accounts' passHref>
                        <Nav.Link>Accounts</Nav.Link> 
                    </Link>
                    <Link href='/portfolio' passHref>
                        <Nav.Link>Portfolio</Nav.Link> 
                    </Link>
                    
                    { 
                        isLoading
                        ? 
                        <Nav>
                            <Spinner animation="border" role="status">
                                <span className="sr-only">Loading...</span>
                            </Spinner>
                        </Nav>
                        :
                        <UserMenu />
                    }

                </Navbar.Collapse>
            </Navbar>
    )
}