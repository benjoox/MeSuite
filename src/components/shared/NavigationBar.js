import React from 'react'
import { useAuth0 } from '@auth0/auth0-react'
import Link from 'next/link'
import { useRouter } from 'next/router'
import { Navbar, Nav, Spinner } from 'react-bootstrap'
import UserMenu from './UserMenu'
import DataMode from './DataMode'

export default function NavigationBar() {
    const { isLoading } = useAuth0()

    return (
        <Navbar bg="dark" expand="lg" className="d-flex">
            <LinkWrapper href="/" passHref>
                <Nav.Link>
                    <h3>Me Suite</h3>
                </Nav.Link>
            </LinkWrapper>
            <Navbar.Toggle aria-controls="navbar-nav" />
            <Navbar.Collapse
                id="navbar-nav"
                className="d-flex justify-content-end"
            >
                <div>
                    <DataMode />
                </div>
                <LinkWrapper href="/accounts" passHref>
                    <Nav.Link>Accounts</Nav.Link>
                </LinkWrapper>
                <LinkWrapper href="/markets" passHref>
                    <Nav.Link>Markets</Nav.Link>
                </LinkWrapper>
                <LinkWrapper href="/calculator" passHref>
                    <Nav.Link>Calculator</Nav.Link>
                </LinkWrapper>

                {isLoading ? (
                    <Spinner animation="border" role="status">
                        <span className="sr-only">Loading...</span>
                    </Spinner>
                ) : (
                    <UserMenu />
                )}
            </Navbar.Collapse>
        </Navbar>
    )
}

const LinkWrapper = ({ href, children }) => {
    const router = useRouter()

    let className = children.props.className || ''
    if (router.pathname === href) {
        className = `${className} selected`
    }

    return (
        <Link href={href} passHref>
            {React.cloneElement(children, { className })}
        </Link>
    )
}
