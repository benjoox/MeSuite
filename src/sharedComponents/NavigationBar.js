import React from 'react'
import { useAuth0 } from "@auth0/auth0-react"
import { Navbar, Nav, Spinner } from 'react-bootstrap'
import UserMenu from './UserMenu'

export default function NavigationBar() {
    const { isLoading } = useAuth0()

    return (
            <Navbar bg="light" expand="lg">
                <Navbar.Brand href="#home">Me </Navbar.Brand>
                <Navbar.Toggle aria-controls="basic-navbar-nav" />
                <Navbar.Collapse id="basic-navbar-nav">
                    <Nav className="mr-auto">
                        <Nav.Link href='#'>
                            Accounts 
                        </Nav.Link>
                        <Nav.Link href='#'>
                            Portfolio 
                        </Nav.Link>
                    </Nav>
                    

                    { 
                        isLoading
                        ? 
                        <Nav>
                            <Spinner animation="border" role="status">
                                <span className="sr-only">Loading...</span>
                            </Spinner>
                        </Nav>
                        :
                        <UserMenu />
                    }

                </Navbar.Collapse>
            </Navbar>
    )
}