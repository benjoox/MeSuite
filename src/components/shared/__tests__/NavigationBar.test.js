import React from 'react'
import renderer from 'react-test-renderer'
// import { useRouter } from 'next/router'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import NavigationBar from '../NavigationBar'

jest.mock('next/router', () => ({
    useRouter() {
        return {
            route: '/',
            pathname: '/accounts',
            query: '',
            asPath: '',
        }
    },
}))
describe('NavigationBar', () => {
    it('should match the snapshot', () => {
        expect.assertions(1)
        const tree = renderer.create(<NavigationBar />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('should have selected classname for the selected link', () => {
        expect.assertions(2)
        const { getByText } = render(<NavigationBar />)

        const accountLink = getByText('Accounts')
        expect(accountLink).toBeInTheDocument()
        expect(accountLink.classList.contains('selected')).toBe(true)
    })
})
