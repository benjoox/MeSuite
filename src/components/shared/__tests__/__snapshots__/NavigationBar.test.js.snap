// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar should match the snapshot 1`] = `
<nav
  className="d-flex navbar navbar-expand-lg navbar-light bg-dark"
>
  <a
    className="nav-link"
    href="/"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseEnter={[Function]}
  >
    <h3>
      Me Suite
    </h3>
  </a>
  <button
    aria-controls="navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="d-flex justify-content-end navbar-collapse collapse"
    id="navbar-nav"
  >
    <div>
      <form
        className=""
        style={
          Object {
            "color": "#007bff",
          }
        }
      >
        <div
          className="custom-control custom-switch"
        >
          <input
            className="custom-control-input"
            disabled={false}
            id="modeSwtich"
            onChange={[Function]}
            secondary={1}
            type="checkbox"
          />
          <label
            className="custom-control-label"
            htmlFor="modeSwtich"
            title=""
          >
            Offline
          </label>
        </div>
      </form>
    </div>
    <a
      className=" selected nav-link"
      href="/accounts"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
    >
      Accounts
    </a>
    <a
      className="nav-link"
      href="/markets"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
    >
      Markets
    </a>
    <a
      className="nav-link"
      href="/calculator"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
    >
      Calculator
    </a>
    <div
      className="spinner-border"
      role="status"
    >
      <span
        className="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</nav>
`;
